### ISSUE TITLE
Grid view iron-list displaying items in incorrect position

### ISSUE THREAD

**Issue by umbertoand** [PolymerElements/iron-list#446]
### Description
Iron-list in grid view sometimes shows list items in incorrect position on the grid. This results in double-ups, items showing out of order and switching behaviour when scrolling. In my particular app this gives me issues as my items need to be sorted in a specific way.

### Expected outcome

Items should appear in grid in the same order that they appear in in the items array

### Actual outcome

Some items appear in grid in different order

### Live Demo
http://jsbin.com/gipovanifu/edit?html,output

### Steps to reproduce
This can be a little tricky to spot in the jsbin but basically keep your eyes on the first column. As each row has 5 items, only items with indices ending in 0 or 5 should appear in this column.

Basically scroll around a bit aggressively until you see any index that isn't a multiple of 5 in the first column. Should look like the screenshot below which I took from the jsbin.

![screen shot 2017-06-05 at 16 21 16](https://cloud.githubusercontent.com/assets/9996705/26790414/8a5045f6-4a0b-11e7-8bc3-da370a74952e.png)

### Browsers Affected
I've been testing on chrome but I've seen it happen on firefox and iOS/desktop safari so far too.

- [ ] Chrome
- [ ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

**Comment by Westbrook** [comment]
I actually just worked a fix for this into my local code. The issue originating from the idea that when the available physical items aren't exactly divisible by the number of items in a row, you can trigger a recycle scenario where items are skipped. To correct this I've added a method that converts non-explicit indexes to the index of last item on that same row, which causes there to be complete rows of items at all times. 

I'm a bit heads down with a big release later this week, but after I'm gonna clean up my changes (update to 2.0) and submit the changes that I made. If you're hurting in the interim you can check out the changes at https://github.com/Westbrook/iron-list/blob/master/iron-list.html#L1809

**Comment by umbertoand** [comment]
Thanks! Great fix, I've incorporated it into our code - lifesaver.

**Comment by jpodwys** [comment]
@Westbrook thanks for this fix! I've incorporated this into my local code and it seems to be helping, but I've noticed that the issue appears to return if I resize my window such that the number of items per row changes. Anyone else seeing this?

**Comment by jpodwys** [comment]
It appears that the issue was introduced after `1.3.x`, as pinning back to `1.3.x` seems to have fixed the issue for me.

**Comment by Westbrook** [comment]
@jpodwys yes that's one of the reasons I've not opened a PR yet, that's a difficulty. It will need to add items to once again match full rows, which can get super tricky over the longer usage of the component. I'll look at your notes on pinning back to `1.3.x` to see if there are any ideas on how to work around that.

**Comment by Westbrook** [comment]
@jpodwys, I'm working this PR together and I'm having trouble getting the resize issue that you're seeing to trigger. Might you be able to get a reliable reproduction situation into a JS bin? I've some thoughts on how I might prevent it from happening, but I'd really like to confirm it to be true in a unit test. Then we can make sure we don't slide around as far as support for this version to version in the future.

@umbertoand I've updated your JS bin with the branch I'm bringing together here: http://jsbin.com/padubilaqe/edit?html,output I'm no longer seeing the offset, which seemed to be starting after scrolling to around the 100th item, but would love to get some extra eyes on this.

**Comment by umbertoand** [comment]
@Westbrook Your updated JS bin appears to be working correctly to my eyes too...

**Comment by jpodwys** [comment]
I just pinned to your fork and I'm still seeing the issue after resize. I'll try to create an isolated demo to determine whether it's my code.

**Comment by jpodwys** [comment]
With a simple jsbin demo, I was unable to reproduce the issue using your PR code.
